crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1603,6 +1604,67 @@ impl<const N: usize> Polygon<N> {
     }
 }
 
+/// A convex polygon with `N` vertices.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect), reflect(Debug, PartialEq))]
+#[cfg_attr(
+    all(feature = "serialize", feature = "bevy_reflect"),
+    reflect(Serialize, Deserialize)
+)]
+pub struct ConvexPolygon<const N: usize> {
+    /// The vertices of the [`ConvexPolygon`].
+    #[cfg_attr(feature = "serialize", serde(with = "super::serde::array"))]
+    pub vertices: [Vec2; N],
```

Comment:
Something to consider: perhaps this should be private, and then we'd have a read-only `vertices` getter method? That way, the convexity can't be invalidated after initial construction, and people also can't manually create a `ConvexPolygon` without the constructors.

Reply:
Good suggestion. I'll do that.